pinot-core/src/main/java/org/apache/pinot/core/segment/virtualcolumn/SingleStringVirtualColumnProvider.java

```diff
@@ -34,9 +33,11 @@
 /**
  * Virtual column provider for a virtual column that contains a single string.
  */
-public abstract class SingleStringVirtualColumnProvider extends BaseVirtualColumnProvider {
+public class SingleStringVirtualColumnProvider extends BaseVirtualColumnProvider {
 
-  protected abstract String getValue(VirtualColumnContext context);
+  protected String getValue(VirtualColumnContext context) {
+    return context.getFieldSpec().getDefaultNullValue().toString();
```

Comment:
This does not seem right to me

Reply:
I know it works, but always returning default null value for SingleStringVirtualColumnProvider is quite confusing.
I think you want to support a virtual column type for default null value, you can add a new class DefaultNullValueVirtualColumnProvider for that purpose.

Reply:
Added a new class DefaultNullValueSingleStringVirtualColumnProvider